Added Label Support, Active Dot Indicator, and Updated Gradle/Dependencies

UmairOye opened the pull request:
This pull request introduces several new features and updates to the curved bottom navigation library:

1. Added Labels: Labels are now available for each navigation item, and users can change the label color via XML.
2. Customizable Active Dot Indicator: An active dot indicator shows the selected item, with options to adjust the dot's size and color.
3. Label Customization: Users can now modify label sizes and colors directly through XML attributes, offering greater flexibility in design.
4. Gradle and SDK Updates: Updated Gradle configuration, target SDK, and compile SDK to version 34. Also updated all dependencies in both the library and the sample project to their latest versions.
5. README Update: Revised the README to include instructions on how to set labels, customize dot sizes, and modify label colors.

These changes aim to enhance the functionality and customizability of the curved bottom navigation view, providing more options for developers to fit their UI requirements.

susonthapa committed:
libraries & gradle updated, labels and active indicator added

UmairOye committed:
Merge remote-tracking branch 'origin/master'

UmairOye committed:
libraries & gradle updated, labels and active indicator added

UmairOye committed:
Update README.md

UmairOye committed:
Update README.md

UmairOye committed:
Update README.md

susonthapa commented:
HI @UmairOye, sorry for the late reply. It has been a while since I worked in this library and I have moved to another stack. I can merge this PR and probably release a new version but I don't know if I will be able to provide support if something goes wrong. If you're willing to maintain the library then I will be more than happy to work together.

UmairOye commented:
A good reply could be:

Yes, I’d be happy to help maintain the library. Let me know how we can proceed from here, and I look forward to working together!

UmairOye commented:
Hi @susonthapa Yes, I’d be happy to help maintain the library. Let me know how we can proceed from here, and I look forward to working together!

UmairOye commented:
@susonthapa any update on this?

susonthapa commented:
> @susonthapa any update on this?

Sorry for the late reply. Life has been super busy. I will try to refresh my memory on how to publish the library 😄.

UmairOye commented:
> > @susonthapa any update on this?
> 
> Sorry for the late reply. Life has been super busy. I will try to refresh my memory on how to publish the library 😄.

@susonthapa All good! I appreciate you taking the time. Let me know if you need any input from my side. Excited to see the update!

susonthapa commented on .idea/compiler.xml:
I was wondering why we are tracking IDE-generated files. Can you please remove these files?

susonthapa commented on build.gradle:
```diff
@@ -1,12 +1,14 @@
 buildscript {
-    ext.kotlin_version = "1.5.0"
+    ext {
+        agp_version = '7.3.1'
```
If we are only using this on line 11, I think it is a bit overkill to extract into a variable. What do you think?

susonthapa commented on curved_bottom_navigation/build.gradle:
```diff
@@ -47,12 +46,14 @@ android {
+    testImplementation 'junit:junit:4.13.2'
+    androidTestImplementation 'androidx.test.ext:junit:1.2.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.6.1'
+    implementation 'com.intuit.sdp:sdp-android:1.1.1'
```
I'm not sure about using this library. I feel like it would be too opinionated to use a scaling library here. I would love to hear your thoughts.

UmairOye replied:
> I'm not sure about using this library. I feel like it would be too opinionated to use a scaling library here. I would love to hear your thoughts.

@susonthapa Thank you for your feedback! I understand your concern about using a scaling library. The reason I included sdp-android is to handle text scaling dynamically across different screen sizes and densities. When using static float values, the text often appears too large or too small on certain devices, which impacts UI consistency.

That said, I'm open to suggestions! If you think there's a better approach to achieve responsive scaling while maintaining flexibility, I'd love to explore that direction. Let me know your thoughts.

susonthapa replied:
@UmairOye I think it is better to leave this to the library's consumer.

susonthapa commented on curved_bottom_navigation/src/main/java/np/com/susanthapa/curved_bottom_navigation/CurvedBottomNavigationView.kt:
```diff
@@ -753,21 +846,49 @@ class CurvedBottomNavigationView @JvmOverloads constructor(
+
+            // If the item is selected, show a dot instead of text
+            if (i == selectedIndex) {
+                canvas.drawCircle(itemCenterX, layoutHeight -12, dotRadius, dotPaint)
```
Do you think it is possible to make the dot optional? I think having a title is nice but about the dot, I'm not sure if everyone wants it.

UmairOye committed:
selected dot added as optional - bug fixed

UmairOye committed:
Removed IDE-generated files and updated .gitignore

UmairOye committed:
Update README.md

readme.md updated

UmairOye commented:
@susonthapa 
I’ve made the requested changes:
-  Added an option to show/hide the active dot based on an XML attribute (app:cbn_showDot). This allows users to customize the UI based on their preference.
- Removed IDE-generated files from the repository and updated .gitignore to prevent them from being tracked in the future.
Let me know if any further modifications are needed. Thanks for the feedback!

UmairOye committed:
sdp dependency removed

UmairOye committed:
Merge remote-tracking branch 'origin/master'

UmairOye commented:
Hi @susonthapa 
I've addressed all the requested changes in this PR, including removing sdp-android and replacing it with a screen-aware scaling extension, as discussed. Just checking in to see if there's anything else you'd like me to adjust or if we’re good to move forward with merging.

Would love to see an update pushed to the library soon — happy to help with anything else needed to get it there. Thanks again for the guidance throughout! 🙌

susonthapa commented:
> Hi @susonthapa I've addressed all the requested changes in this PR, including removing sdp-android and replacing it with a screen-aware scaling extension, as discussed. Just checking in to see if there's anything else you'd like me to adjust or if we’re good to move forward with merging.
> 
> Would love to see an update pushed to the library soon — happy to help with anything else needed to get it there. Thanks again for the guidance throughout! 🙌

Thanks @UmairOye for fixing the issue. I will test this and merge the PR if everything works. I will let you know when it is ready to be published.

susonthapa commented:
@UmairOye I tested the code and it works well. I think it would be better if we could add animations to the title and dot but that is something we can look at later. I will be merging the PR. 

https://github.com/user-attachments/assets/b26b3899-08f1-4709-94f4-33599f3c82f7
